controllers/configurationpolicy_controller.go

```diff
@@ -1046,8 +1042,7 @@ func getDetails(unstruct unstructured.Unstructured) (name string, kind string, n
 func buildNameList(
 	unstruct unstructured.Unstructured, complianceType string, resList *unstructured.UnstructuredList,
 ) (kindNameList []string) {
-	for i := range resList.Items {
```

Comment:
In the original code, `uObj` is the original struct but the for range I believe creates a copy. This would be totally fine except it seems that `handleSingleKey` is passed the pointer of `uObj` so is `handleSingleKey` modifying `uObj` and is the caller to `buildNameList` expecting its argument of `resList` to be modified?

Reply:
These comments about pointers and copies are giving me flashbacks to my Advent of Code stuff with Rust 😓 

Just looking at the code around this change, I think you are correct in general, the `range` makes a copy, which means I may have altered the behavior. I think in another branch of refactoring that I threw away, I changed this to not be a pointer at all and it still worked. So I don't think we use the mutability of the original thing (as in, the copy is fine). BUT making an unnecessary copy of an entire `unstruct` could be getting into the realm where passing a pointer is better for performance.

tl;dr I will revert 😆